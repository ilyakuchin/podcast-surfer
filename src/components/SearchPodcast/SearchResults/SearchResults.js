import React, { Component } from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

const Podcasts = styled.ul`
  margin: 0;
  padding: 0;
  margin-top: 50px;
  grid-area: results;
  list-style-type: none;
`;

const PodcastGrid = styled.li`
  margin-top: 50px;
  justify-items: center;
  align-items: center;
  max-width: 700px;
  display: grid;
  grid-template-areas: "podcast-image podcast-link";
  justify-items: center;
  align-items: center;
  grid-template-columns: 200px auto;

  @media (max-width: 600px) {
    grid-template-areas:
      "podcast-image"
      "podcast-link";
  }
`;

const PodcastImage = styled.img`
  grid-area: podcast-image;
  width: 200px;
  height: 200px;
  object-fit: scale-down;
`;

const PodcastLink = styled.div`
  grid-area: podcast-link;

  @media (max-width: 600px) {
    text-align: center;
  }
`;

export default class SearchResults extends Component {
  render() {
    return (
      <Podcasts>
        {this.props.podcasts.map(item => (
          <PodcastGrid key={item.id}>
            <PodcastImage src={item.image} alt="podcast cover" />
            <PodcastLink>
              <Link to={{ pathname: "/podcast", state: { rss: item.rss } }}>
                {item.name}
              </Link>
            </PodcastLink>
          </PodcastGrid>
        ))}
      </Podcasts>
    );
  }
}
