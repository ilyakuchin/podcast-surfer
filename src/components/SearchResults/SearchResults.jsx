import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchCurrentPodcast } from '../../redux/actions/CurrentPodcast/currentPodcast';
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
import { fetchPopularPodcasts } from '../../redux/actions/Podcasts/podcasts';

const Podcasts = styled.ul`
  margin: 0;
  padding: 0;
  margin-top: 50px;
  grid-area: results;
  list-style-type: none;
`;

const PodcastGrid = styled.li`
  margin-top: 50px;
  justify-items: center;
  align-items: center;
  max-width: 700px;
  display: grid;
  grid-template-areas: 'podcast-image podcast-link';
  justify-items: center;
  align-items: center;
  grid-template-columns: 200px auto;

  @media (max-width: 600px) {
    grid-template-areas:
      'podcast-image'
      'podcast-link';
  }
`;

const PodcastImage = styled.img`
  grid-area: podcast-image;
  width: 200px;
  height: 200px;
  object-fit: scale-down;
`;

const CenterSpinner = styled.div`
  display: grid;
  grid-area: results;
  width: 100%;
  height: 100%;
  justify-items: center;
  align-items: center;
`;

const PodcastLink = styled.div`
  grid-area: podcast-link;

  @media (max-width: 600px) {
    text-align: center;
  }
`;

export function SearchResults({
  isFetching,
  jwt,
  podcasts,
  fetchCurrentPodcastConnect,
  fetchPopularPodcasts
}) {
  useEffect(() => {
    fetchPopularPodcasts();
  }, [fetchPopularPodcasts]);
  return !isFetching ? (
    <Podcasts>
      {podcasts.map(({ id, name, image, rss }) => (
        <PodcastGrid key={id}>
          <PodcastImage src={image} alt='podcast cover' />
          <PodcastLink>
            <Link
              onClick={() => fetchCurrentPodcastConnect(rss, jwt)}
              to='/podcast'
            >
              {name}
            </Link>
          </PodcastLink>
        </PodcastGrid>
      ))}
    </Podcasts>
  ) : (
    <CenterSpinner>
      <LoadingSpinner />
    </CenterSpinner>
  );
}

SearchResults.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  jwt: PropTypes.string.isRequired,
  podcasts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      image: PropTypes.string,
      rss: PropTypes.string,
      name: PropTypes.string
    })
  ).isRequired,
  fetchCurrentPodcastConnect: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    jwt: state.userInfo.jwt,
    podcasts: state.podcasts.podcasts,
    isFetching: state.podcasts.isFetching
  };
};

const dispatchToProps = dispatch => {
  return {
    fetchCurrentPodcastConnect: (rss, jwt) =>
      dispatch(fetchCurrentPodcast(rss, jwt)),
    fetchPopularPodcasts: () => dispatch(fetchPopularPodcasts())
  };
};

const ConnectedSearchResults = connect(
  mapStateToProps,
  dispatchToProps
)(SearchResults);

export default ConnectedSearchResults;
