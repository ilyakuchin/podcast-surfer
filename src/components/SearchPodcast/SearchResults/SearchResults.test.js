import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import SearchResults from "./SearchResults";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("Test SearchResults Component", () => {
  test("should render correctly", () => {
    const podcasts = [
      { id: 1, image: "link", rss: "link", name: "name" },
      { id: 2, image: "link2", rss: "link2", name: "name2" },
      { id: 3, image: "link3", rss: "link3", name: "name3" }
    ];

    const wrapper = shallow(<SearchResults podcasts={podcasts} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
