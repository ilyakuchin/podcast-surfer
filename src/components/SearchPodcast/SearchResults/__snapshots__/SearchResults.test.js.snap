// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchResults Component should render correctly 1`] = `
<styled.ul>
  <styled.li
    key="1"
  >
    <styled.img
      alt="podcast cover"
      src="link"
    />
    <styled.div>
      <Link
        to={
          Object {
            "pathname": "/podcast",
            "state": Object {
              "rss": "link",
            },
          }
        }
      >
        name
      </Link>
    </styled.div>
  </styled.li>
  <styled.li
    key="2"
  >
    <styled.img
      alt="podcast cover"
      src="link2"
    />
    <styled.div>
      <Link
        to={
          Object {
            "pathname": "/podcast",
            "state": Object {
              "rss": "link2",
            },
          }
        }
      >
        name2
      </Link>
    </styled.div>
  </styled.li>
  <styled.li
    key="3"
  >
    <styled.img
      alt="podcast cover"
      src="link3"
    />
    <styled.div>
      <Link
        to={
          Object {
            "pathname": "/podcast",
            "state": Object {
              "rss": "link3",
            },
          }
        }
      >
        name3
      </Link>
    </styled.div>
  </styled.li>
</styled.ul>
`;
